test/server.test.js

```diff
@@ -2051,6 +2051,34 @@ test(
     }
 );
 
+// eslint-disable-next-line max-len
+test("should emit 'after' on client closed request with an uncaughtException", function(t) {
+    SERVER.on('after', function(req, res, route, err) {
+        t.ok(err);
+        t.equal(req.connectionState(), 'close');
+        t.equal(res.statusCode, 444);
+        t.equal(err.name, 'Error');
+        t.end();
+    });
+
+    SERVER.on('uncaughtException', function(req, res, route, err, next) {
+        next();
+    });
+
+    SERVER.get('/foobar', function(req, res, next) {
+        // fast client times out at 500ms, wait for 800ms which should cause
+        // client to timeout
+        setTimeout(function() {
+            throw new Error('foo');
+        }, 800);
```

Comment:
Can we rely on a 'timeout' event on `FAST_CLIENT` instead of a timer duration to make sure that we throw after the client times out and abort the request? Or maybe even relying on an `abort` event on the client would be better? Or even better on the 'abort' event of the request object?

Otherwise it seems this test could be broken easily by changes in the client's implementation details.

Reply:
Could do, I mostly copied this pattern from existing tests so I'm not sure if you'd prefer consistency with those tests over stability for this test?

Reply:
I prefer stability here. Your stable test can act as a good example of how to correctly write tests and that pattern can then be used for future tests. I think changing the other tests to follow your pattern of stable tests can also be a good first issue for new contributors.

Reply:
@cprussin @rajatkumar I refactored the test so that it doesn't rely on any arbitrary timeout, but instead relies on the actual state of the request/response lifecycle. Let me know if that looks good to you!